src/tools/genpolicy/src/secret.rs

```diff
@@ -33,6 +33,9 @@ pub struct Secret {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     immutable: Option<bool>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    r#type: Option<String>,
```

Comment:
Sorry to keep asking you stupid questions, but what happens if not a raw type `r#type` like `data` or `immutable`?

IIUC it will ignore the `type: Opaque` on pod-nested-configmap-secret.yaml's Secret:


```
<snip>
---
apiVersion: v1
data:
  secret_key: bXlzZWNyZXQ= #mysecret
kind: Secret
metadata:
  name: secret
type: Opaque
---
<snip>
```

Reply:
Please never worry about "stupid" questions with me - I am not a rust expert either.

Here I believe "r#type" is needed here because "type" is a reserved keyword in Rust. We're forcing the compiler to use "type" as a struct field name here.

Reply:
When I said "Ignore the optional type field" I meant:

- If serde_yaml finds the type field in an intput YAML file, it gives that field value to genpolicy, and *genpolicy will ignore it*
- If serde_yaml doesn't find the type field in an intput YAML file, it gives None as the field value to genpolicy, and *genpolicy will ignore None*

In other words, this new r#type behaves the same way as the new nodeName field from https://github.com/kata-containers/kata-containers/commit/076869aa39d96b548aa1ab4c12c999239392e823.